cmd/clusterctl/examples/google/machine_setup_configs.yaml

```diff
@@ -28,30 +28,32 @@ items:
           chmod $3 $2
       }
 
-      curl -sf https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
-      touch /etc/apt/sources.list.d/kubernetes.list
-      sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
       apt-get update -y
-      apt-get install -y \
-        socat \
-        ebtables \
-        apt-transport-https \
-        cloud-utils \
-        prips
+      apt-get install -y socat ebtables apt-get install apt-transport-https ca-certificates curl software-properties-common cloud-utils prips
+      curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+      cat <<EOF > /etc/apt/sources.list.d/k8s.list
+      deb [arch=amd64] https://download.docker.com/linux/ubuntu ubuntu-xenial stable
+      EOF
+      apt-get update -y
 
       function install_configure_docker () {
-        # prevent docker from auto-starting
-        echo "exit 101" > /usr/sbin/policy-rc.d
-        chmod +x /usr/sbin/policy-rc.d
-        trap "rm /usr/sbin/policy-rc.d" RETURN
-        apt-get install -y docker.io
-        echo 'DOCKER_OPTS="--iptables=false --ip-masq=false"' > /etc/default/docker
-        systemctl daemon-reload
-        systemctl enable docker
-        systemctl start docker
+          # prevent docker from auto-starting
+          echo "exit 101" > /usr/sbin/policy-rc.d
+          chmod +x /usr/sbin/policy-rc.d
+          trap "rm /usr/sbin/policy-rc.d" RETURN
+          apt-get install -y docker-ce=18.06.2~ce~3-0~ubuntu
```

Comment:
I do wonder if we should just use the tar.gz packaging - the docker packages seem to cause as many problems as they solve.  We support it in kops, but only do it for OSes where docker doesn't package.  But when we go to containerd, I think we're going to try switching.

Reply:
I was mainly trying to patch things up enough to get them working again without too much disruption :)